### PR TITLE
set up scaffolding for js tests

### PR DESCRIPTION
**Pull request by devvmh**
@harlantwood @Connoropolous please review 😄

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,10 +5,10 @@
 #   git config --global core.excludesfile ~/.gitignore_global
 
 #assety stuff
-realtime/node_modules
 public/assets
 public/metamaps_mobile
 vendor/
+node_modules
```
Comment (Connoropolous) [inline .gitignore]: Will this ignore node modules folder in both directories?

**Comment by Connoropolous** [comment]
Looks good to me Devin!

File: .gitignore (hunk 1 above)
Comment (harlantwood) [inline .gitignore]: Yes

File: app/assets/javascripts/README-testing-js.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Change directories to where this file is, and then run 
+
+    npm install
+
+to set up your testing environment. Then use
+
+    npm run test
```
Comment (harlantwood) [inline app/assets/javascripts/README-testing-js.md]: `npm test` works too and is the usual incantation :)

File: app/assets/javascripts/package.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "metamaps-frontend",
+  "version": "1.0.0",
+  "description": "Metamaps frontend - currently just tests",
+  "scripts": {
+    "test": "mocha test || echo 'Run `npm install` to setup testing'"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/metamaps/metamaps.git"
```
Comment (harlantwood) [inline app/assets/javascripts/package.json]: All URLs should point to existing repo

**Comment by harlantwood** [comment]
Looks good @devvmh!  Could be another PR, but let's make sure to run in CI as well.

**Comment by devvmh** [comment]
Thanks guys! I got it hooked into Travis. Woohoo!

**Comment by devvmh** [comment]
Oh yeah, and this should reference #503

**Comment by harlantwood** [comment]
Awesome!  https://travis-ci.org/metamaps/metamaps/builds/125729999

**Pull request merged**